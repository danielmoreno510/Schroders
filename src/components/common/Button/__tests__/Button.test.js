import React from 'react';
import TestRenderer from 'react-test-renderer';

import Button from '../Button';

describe('Button component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      primaryColor: '#002337',
      onPress: jest.fn(),
    };

    describe('WHEN the component is rendered without props', () => {
      const testRenderer = TestRenderer.create(<Button {...testProps} />);

      it('THEN should display a regular button', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
      });
    });

    describe('WHEN the component is rendered with props', () => {
      const customProps = {
        ...testProps,
        text: 'Test',
        disabled: true,
        styleButton: {color: 'red'},
        styleText: {color: 'red'},
      };
      const testRenderer = TestRenderer.create(<Button {...customProps} />);

      it('THEN should display a regular button', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
      });
    });
  });
});
