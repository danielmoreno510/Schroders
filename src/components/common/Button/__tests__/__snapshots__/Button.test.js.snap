// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component GIVEN a set of valid props WHEN the component is rendered with props THEN should display a regular button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f1b30a",
      "color": "red",
      "height": 56,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
        },
        {
          "color": "red",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`Button component GIVEN a set of valid props WHEN the component is rendered without props THEN should display a regular button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f1b30a",
      "height": 56,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
/>
`;
